src/main/java/edu/ie3/datamodel/utils/ValidationUtils.java

```diff
@@ -231,4 +260,172 @@ public static boolean checkMeasurementUnit(MeasurementUnitInput measurementUnit)
+        rawGridElements.getTransformer3Ws().stream()
+            .map(
+                transformer -> {
+                  if (!nodes.contains(transformer.getNodeA())
+                      || !nodes.contains(transformer.getNodeB())
+                      || !nodes.contains(transformer.getNodeC()))
+                    throw getMissingNodeException(transformer);
+                  return checkTransformer3W(transformer);
+                })
+            .anyMatch(cond -> !cond);
+    if (anyNullTransformer3w)
+      throw new InvalidGridException(
+          "The list of three winding transformers contains at least one NULL element.");
+
+    /* Checking switches */
+    boolean anyNullSwitch =
+        rawGridElements.getSwitches().stream()
+            .map(switcher -> checkSwitch(switcher) && checkNodeAvailability(switcher, nodes))
+            .anyMatch(cond -> !cond);
+    if (anyNullSwitch)
+      throw new InvalidGridException("The list of switches contains at least one NULL element.");
+
+    /* Checking measurement units */
+    boolean anyNullMeasurement =
+        rawGridElements.getMeasurementUnits().stream()
+            .map(
+                measurement -> {
+                  if (!nodes.contains(measurement.getNode()))
+                    throw getMissingNodeException(measurement);
+                  return checkMeasurementUnit(measurement);
+                })
+            .anyMatch(cond -> !cond);
+    if (anyNullMeasurement)
+      throw new InvalidGridException(
+          "The list of measurement units contains at least one NULL element.");
+
+    return true;
+  }
+
+  /**
+   * Checks the validity of each and every system participant. Moreover, it checks, if the systems
+   * are connected to an node that is not in the provided set
+   *
+   * @param systemParticipants The system participants
+   * @param nodes Set of already known nodes
+   * @return true
+   */
+  public static boolean checkSystemParticipants(
+      SystemParticipants systemParticipants, Set<NodeInput> nodes) {
+    if (systemParticipants == null) return false;
+
+    systemParticipants.getBmPlants().forEach(entity -> checkNodeAvailability(entity, nodes));
+
+    systemParticipants.getChpPlants().forEach(entity -> checkNodeAvailability(entity, nodes));
+
+    /* TODO: Electric vehicle charging systems are currently only dummy implementation */
+
+    systemParticipants.getFixedFeedIns().forEach(entity -> checkNodeAvailability(entity, nodes));
+
+    systemParticipants.getHeatPumps().forEach(entity -> checkNodeAvailability(entity, nodes));
+
+    systemParticipants.getLoads().forEach(entity -> checkNodeAvailability(entity, nodes));
+
+    systemParticipants.getPvPlants().forEach(entity -> checkNodeAvailability(entity, nodes));
+
+    systemParticipants.getStorages().forEach(entity -> checkNodeAvailability(entity, nodes));
+
+    systemParticipants.getWecPlants().forEach(entity -> checkNodeAvailability(entity, nodes));
```

Comment:
```suggestion
 systemParticipants.allEntitiesAsList().parallelStream().forEach(entity -> checkNodeAvailability((SystemParticipantInput)entity, nodes));
```

Reply:
Doesn't work as `EvcsInput` is no child of `SystemParticipantInput`.